storagenode/piecestore/endpoint.go

```diff
@@ -55,6 +56,7 @@ type OldConfig struct {
 // Config defines parameters for piecestore endpoint.
 type Config struct {
 	ExpirationGracePeriod time.Duration `help:"how soon before expiration date should things be considered expired" default:"48h0m0s"`
+	MaxConcurrentRequests int           `help:"how many concurrent requests are allowed, before uploads are rejected." default:"30"`
```

Comment:
How would SNOs know what value to configure here?

Is there a way for the storage node to self-diagnose and determine if it is overloaded or not?

Reply:
Probably with some kind of benchmarking by some other party. The issue is that it's not just about the storage node itself, but also about the network it's able to deliver.

I guess we could try monitoring bandwidth usage and when it isn't increasing anymore or load or memory usage etc. But these all are much more complicated solutions than "this is how much this storage node can serve".

Reply:
I think it’s totally fine to have a default, if someone notices a lot of issues he just can tune it down. We had the same mechanic in V2 and that worked fine.
About the default itself, I would recommend something in the neighborhood of 5-10 max.

Reply:
Yeah, I'm not sure what to put as limits for now. Is there an easy way to test how much we can handle?

Reply:
My Server Node was handling 15-20 Requests max per second, when fully cached.
The Pi 3B+ is already overwhelmed with 3 requests.

Reply:
I could run a few test uploads on my local Network to see what i can come up with.

Reply:
I wonder if the storagenodes could potentially obtain data from one or more satellites to help it decide how many requests it can handle and adjust the max.

Reply:
With a dynamic scaling that is just adding another layer of potential issues @phutchins 
Lets make a good decision/average and update the docs accordingly

Reply:
Let me know when you get a number from testing @stefanbenten

Reply:
I will test that after lunch 👍